File: src/cli/commands/workspace.js
import chalk from 'chalk';
import fs from 'node:fs';
import path from 'node:path';
import { addToWorkspace, removeFromWorkspace, listWorkspace } from '../../lib/workspace.js';

export function registerWorkspaceCommand(program) {
  const ws = program
    .command('workspace')
    .alias('ws')
    .description('Manage registered projects across your machine');

  ws.command('add [path]')
    .description('Register a project in the workspace')
    .option('--alias <alias>', 'Short name for the project')
    .action((projectPath, opts) => {
      const resolved = path.resolve(projectPath || '.');
      if (!fs.existsSync(path.join(resolved, '.projctl', 'config.json'))) {
        console.error(chalk.red(`No .projctl/ found at ${resolved}. Run \`projctl init\` first.`));
        process.exit(1);
      }
      const alias = opts.alias || path.basename(resolved);
      addToWorkspace(resolved, alias);
      console.log(chalk.green(`Added ${alias} → ${resolved}`));
    });

  ws.command('remove <alias>')
    .description('Unregister a project')
    .action((alias) => {
      removeFromWorkspace(alias);
      console.log(chalk.green(`Removed ${alias}`));
    });

  ws.command('list')
    .alias('ls')
    .description('List all registered projects')
    .action(() => {
      const projects = listWorkspace();
      if (projects.length === 0) {
        console.log(chalk.dim('No projects registered. Run `projctl init` in a project directory.'));
        return;
      }
      for (const p of projects) {
        const exists = fs.existsSync(path.join(p.path, '.projctl', 'config.json'));
        const status = exists ? chalk.green('●') : chalk.red('●');
        console.log(`  ${status} ${chalk.bold(p.alias.padEnd(20))} ${chalk.dim(p.path)}`);
      }
    });
}

File: projctl/cli/workspace.py
from __future__ import annotations
from pathlib import Path
from typing import Optional

import typer
from ._console import console

from ..lib.workspace import add_to_workspace, remove_from_workspace, list_workspace

app = typer.Typer(help="Manage registered projects across your machine")


@app.command("add")
def workspace_add(
    project_path: Optional[str] = typer.Argument(None, help="Path to the project (default: current dir)"),
    alias: Optional[str] = typer.Option(None, "--alias", help="Short name for the project"),
):
    """Register a project in the workspace."""
    resolved = Path(project_path or ".").resolve()
    if not (resolved / ".projctl" / "config.json").exists():
        console.print(f"[red]No .projctl/ found at {resolved}. Run `projctl init` first.[/red]")
        raise typer.Exit(1)
    name = alias or resolved.name
    add_to_workspace(resolved, name)
    console.print(f"[green]Added {name} → {resolved}[/green]")


@app.command("remove")
def workspace_remove(alias: str = typer.Argument(...)):
    """Unregister a project."""
    remove_from_workspace(alias)
    console.print(f"[green]Removed {alias}[/green]")


@app.command("list")
def workspace_list():
    """List all registered projects."""
    projects = list_workspace()
    if not projects:
        console.print("[dim]No projects registered. Run `projctl init` in a project directory.[/dim]")
        return
    for p in projects:
        exists = (Path(p["path"]) / ".projctl" / "config.json").exists()
        status = "[green]●[/green]" if exists else "[red]●[/red]"
        console.print(f"  {status} [bold]{p['alias']:<20}[/bold] [dim]{p['path']}[/dim]")
